crypto/fipsmodule/bn/montgomery.c

```diff
@@ -437,12 +448,80 @@ int BN_mod_mul_montgomery(BIGNUM *r, const BIGNUM *a, const BIGNUM *b,
     // This bound is implied by |bn_mont_ctx_set_N_and_n0|. |bn_mul_mont|
     // allocates |num| words on the stack, so |num| cannot be too large.
     assert((size_t)num <= BN_MONTGOMERY_MAX_WORDS);
+
+#if defined(BN_MONTGOMERY_USE_S2N_BIGNUM)
```

Comment:
this should be factored out and also made run-time conditions. Can get inspiration from x25519 work.

Reply:
Thanks, I will check out pull requests for x25519 in the past.

Reply:
Thanks, I will check out pull requests for x25519 in the past.